clang/test/SemaSYCL/decomposition.cpp

```diff
@@ -77,13 +86,13 @@ int main() {
     myQueue.submit([&](sycl::handler &h) {
       h.single_task<class Acc1>([=]() { return t1.i; });
     });
-    // CHECK: FunctionDecl {{.*}}Acc1{{.*}} 'void (__global char *, sycl::range<1>, sycl::range<1>, sycl::id<1>, __global char *, sycl::range<1>, sycl::range<1>, sycl::id<1>, __global char *, sycl::range<1>, sycl::range<1>, sycl::id<1>, StructNonDecomposed, int)'
+    // CHECK: FunctionDecl {{.*}}Acc1{{.*}} 'void (__global char *, sycl::range<1>, sycl::range<1>, sycl::id<1>, __wrapper_class, __global char *, sycl::range<1>, sycl::range<1>, sycl::id<1>, __wrapper_class, __global char *, sycl::range<1>, sycl::range<1>, sycl::id<1>, __wrapper_class, StructNonDecomposed, int)'
```

Comment:
Just for my education, there is wrapper classes added because StructWithArray contains **array** of structs with accessor and decomposition for arrays still happens, right?
Not sure that I fully understand the purpose of adding a pointer to `StructWithAccessor` since accessor triggers decomposition anyway.

Reply:
I thought I would add a test to ensure that cases where accessors + pointers are in class works as expected - i.e. triggers decomposition.

Reply:
Ok, makes sense.